src/controllers/couponCodeController.js

```diff
@@ -0,0 +1,177 @@
+/* eslint-disable*/
+import { Op, where } from "sequelize";
+import models from "../database/models";
+import generateCouponCode from "../utils/helpers/couponGenerate";
+
+const { CouponCodeDiscount, Product, Vendor, Buyer, Cart } = models;
+
+class CouponCodeController {
+  static async createCouponCode(req, res) {
+    const { productId, discount, expirationDate, maxUsage } = req.body;
+    if (discount <= 0) {
+      return res
+        .status(409)
+        .json({
+          status: "fail",
+          message: "dicount must be greater than zero."
+        });
+    }
+    const couponCode = generateCouponCode();
+    const vendor = await Vendor.findOne({ where: { UserId: req.user.id } });
+    const newCoupon = {
+      couponCode,
+      discount,
+      expirationDate,
+      maxUsage,
+      product: productId
+    };
+    try {
+      const coupon = await CouponCodeDiscount.create(newCoupon);
+      await coupon.setVendor(vendor);
+      return res.status(201).json({ status: "success", data: coupon });
+    } catch (error) {
+      return res.status(500).json({ status: "fail", error: error.message });
+    }
+  }
+
+  static async getVendorCoupons(req, res) {
+    try {
+      const vendor = await Vendor.findOne({ where: { UserId: req.user.id } });
+      const vendorCoupons = await CouponCodeDiscount.findAll({
+        where: { VendorId: vendor.dataValues.id }
+      });
+      return res.status(200).json({ status: "succcess", data: vendorCoupons });
+    } catch (error) {
+      return res.status(500).json({ status: "fail", error: error.message });
+    }
+  }
+
+  static async checkExpiredCoupons(req, res) {
```

Comment:
- the function name `checkExpiredCoupons` gives the image of oly checking while you are doing more in it which is disabling discount
- you can seperate the two may or keep it in one function and you change the name.